yt_dlp/extractor/dailymotion.py

```diff
@@ -383,6 +692,37 @@ def _extract_embed_urls(cls, url, webpage):
                 yield '//dailymotion.com/playlist/%s' % p
 
 
+class DailymotionSearchIE(DailymotionPlaylistBaseIE):
+    IE_NAME = 'dailymotion:search'
+    _VALID_URL = r'(?:https?://)?(?:www\.)?dailymotion\.[a-z]{2,3}/search/(?P<id>[^/]+)/videos'
+    _PAGE_SIZE = 20
+    _TESTS = [{
+        'url': 'http://www.dailymotion.com/search/king of turtles/videos',
+        'info_dict': {
+            'id': 'king of turtles',
+            'title': 'Dailymotion search of king of turtles',
+        },
+        'playlist_mincount': 90,
+    }]
+
+    def _fetch_page(self, term, page):
+        page += 1
+        videos = self._call_search_api(term, page, 'Searching "%s", page %d' % (term, page))['videos']
+        for edge in videos['edges']:
+            node = edge['node']
+            yield self.url_result(
+                'https://www.dailymotion.com/video/' + node['xid'],
+                DailymotionIE.ie_key(),
+                node['xid']
+            )
```

Comment:
Should be non-fatal:
```suggestion
        response = self._call_search_api(term, page, f'Searching "{term}" page {page}')
        for edge in traverse_obj(response, ('videos', 'edges', lambda _, x: x['node']['xid'])) or []:
            xid = edge['node']['xid']
            yield self.url_result(f'https://www.dailymotion.com/video/{xid}', DailymotionIE, xid)
```